pkg/resource/function/hooks.go

```diff
@@ -491,55 +491,69 @@ func (rm *resourceManager) updateFunctionConcurrency(
+			FunctionName: aws.String(*desired.ko.Spec.Name),
+		}
+		_, err = rm.sdkapi.DeleteFunctionEventInvokeConfigWithContext(ctx, input_delete)
+		rm.metrics.RecordAPICall("DELETE", "DeleteFunctionEventInvokeConfig", err)
+		if err != nil {
+			return err
+		}
+		return nil
+	}
+
 	dspec := desired.ko.Spec
 	input := &svcsdk.PutFunctionEventInvokeConfigInput{
 		FunctionName: aws.String(*dspec.Name),
 	}
 
-	if desired.ko.Spec.FunctionEventInvokeConfig != nil {
-		if desired.ko.Spec.FunctionEventInvokeConfig.MaximumEventAgeInSeconds != nil {
-			input.MaximumEventAgeInSeconds = aws.Int64(*desired.ko.Spec.FunctionEventInvokeConfig.MaximumEventAgeInSeconds)
-		}
-		if desired.ko.Spec.FunctionEventInvokeConfig.MaximumRetryAttempts != nil {
-			input.MaximumRetryAttempts = aws.Int64(*desired.ko.Spec.FunctionEventInvokeConfig.MaximumRetryAttempts)
-		}
-		if desired.ko.Spec.FunctionEventInvokeConfig.DestinationConfig != nil {
-			destinations := &svcsdk.DestinationConfig{}
-			if desired.ko.Spec.FunctionEventInvokeConfig.DestinationConfig.OnFailure != nil {
-				destinations.OnFailure = &svcsdk.OnFailure{}
-				if desired.ko.Spec.FunctionEventInvokeConfig.DestinationConfig.OnFailure.Destination != nil {
-					destinations.OnFailure.Destination = aws.String(*desired.ko.Spec.FunctionEventInvokeConfig.DestinationConfig.OnFailure.Destination)
-				}
+	if dspec.FunctionEventInvokeConfig.DestinationConfig != nil {
+		destinations := &svcsdk.DestinationConfig{}
+		if dspec.FunctionEventInvokeConfig.DestinationConfig.OnFailure != nil {
+			destinations.OnFailure = &svcsdk.OnFailure{}
+			if dspec.FunctionEventInvokeConfig.DestinationConfig.OnFailure.Destination != nil {
+				destinations.OnFailure.Destination = aws.String(*dspec.FunctionEventInvokeConfig.DestinationConfig.OnFailure.Destination)
 			}
-			if desired.ko.Spec.FunctionEventInvokeConfig.DestinationConfig.OnSuccess != nil {
-				destinations.OnSuccess = &svcsdk.OnSuccess{}
-				if desired.ko.Spec.FunctionEventInvokeConfig.DestinationConfig.OnSuccess.Destination != nil {
-					destinations.OnSuccess.Destination = aws.String(*desired.ko.Spec.FunctionEventInvokeConfig.DestinationConfig.OnSuccess.Destination)
-				}
+		}
+		if dspec.FunctionEventInvokeConfig.DestinationConfig.OnSuccess != nil {
+			destinations.OnSuccess = &svcsdk.OnSuccess{}
+			if dspec.FunctionEventInvokeConfig.DestinationConfig.OnSuccess.Destination != nil {
+				destinations.OnSuccess.Destination = aws.String(*dspec.FunctionEventInvokeConfig.DestinationConfig.OnSuccess.Destination)
 			}
-			input.DestinationConfig = destinations
 		}
+		input.DestinationConfig = destinations
+	}
+	if dspec.FunctionEventInvokeConfig.MaximumEventAgeInSeconds != nil {
+		input.MaximumEventAgeInSeconds = aws.Int64(*dspec.FunctionEventInvokeConfig.MaximumEventAgeInSeconds)
+	}
+	if dspec.FunctionEventInvokeConfig.MaximumRetryAttempts != nil {
+		input.MaximumRetryAttempts = aws.Int64(*dspec.FunctionEventInvokeConfig.MaximumRetryAttempts)
 	}
```

Comment:
why are these handled outside of the top if statement in L524

Reply:
Because these two `MaximumEventInvokeConfig` and `MaximumRetryAttempts` are another fields in `FunctionEventInvokeConfig` and not a part of `DestinationConfig` field.